CMakeLists.txt

```diff
@@ -264,33 +273,32 @@ find_opt_library_and_header(EDIT_PATH histedit.h EDIT_LIB edit HAVE_EDIT)
 ########################################################################
 
 macro(add_ledger_library_dependencies _target)
-  target_link_libraries(${_target} ${MPFR_LIB})
-  target_link_libraries(${_target} ${GMP_LIB})
+  target_link_libraries(${_target} PRIVATE ${MPFR_LIB} ${GMP_LIB})
   if (HAVE_EDIT)
-    target_link_libraries(${_target} ${EDIT_LIB})
+    target_link_libraries(${_target} PUBLIC ${EDIT_LIB})
   endif()
   if (HAVE_GETTEXT)
-    target_link_libraries(${_target} ${INTL_LIB})
+    target_link_libraries(${_target} PRIVATE ${INTL_LIB})
   endif()
   if (HAVE_GPGME)
-    target_link_libraries(${_target} Gpgmepp)
+    target_link_libraries(${_target} PRIVATE Gpgmepp)
   endif()
   if (HAVE_BOOST_PYTHON)
     if (CMAKE_SYSTEM_NAME STREQUAL Darwin)
       # Don't link directly to a Python framework on macOS, to avoid segfaults
       # when the module is imported from a different interpreter
-      target_link_libraries(${_target} ${Boost_LIBRARIES})
+      target_link_libraries(${_target} PUBLIC ${Boost_LIBRARIES})
       set_target_properties(${_target} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
     else()
-      target_link_libraries(${_target} ${Boost_LIBRARIES} ${Python_LIBRARIES})
+      target_link_libraries(${_target} PUBLIC ${Boost_LIBRARIES} ${Python_LIBRARIES})
```

Comment:
ℹ️ The `${Boost_LIBRARIES}` need to be public as the Util and Math tests use it and the linker will fail otherwise.